pkg/controller/keyvault/spc/reconcile_test.go

```diff
@@ -0,0 +1,976 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package spc
+
+import (
+	"context"
+	"fmt"
+	"iter"
+	"testing"
+
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/controllername"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/metrics"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/testutils"
+	"github.com/Azure/aks-app-routing-operator/pkg/manifests"
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	secv1 "sigs.k8s.io/secrets-store-csi-driver/apis/v1"
+)
+
+// Test the Reconcile method with successful case
+func TestReconcileSuccess(t *testing.T) {
+	scheme := runtime.NewScheme()
```

Comment:
can we make this a global across the test file + use a testMain or maybe even make registration a single function that each test calls?

Reply:
it's just a decent bit of repeated code but i'm not gonna block on this